R/seek_variables.R

```diff
@@ -0,0 +1,169 @@
+#' @title Find variables by its name, variable or value labels
+#' @name seek_variables
+#'
+#' @description This functions seeks variables in a data frame, based on patterns
+#' that either match the variable name (column name), variable labels, value labels
+#' or factor levels. Matching variable and value labels only works for "labelled"
+#' data, i.e. when the variables either have a `label` attribute or `labels`
+#' attribute.
+#'
+#' `seek_variables()` is particular useful for larger data frames with labelled
+#' data - finding the correct variable name can be a challenge. This function
+#' helps to find the required variables, when only certain patterns of variable
+#' names or labels are known.
+#'
+#' @param data A data frame.
+#' @param pattern Character string (regular expression) to be matched in `data`.
+#' May also be a character vector of length > 1. `pattern` is searched for in
+#' column names, variable label and value labels attributes, or factor levels of
+#' variables in `data`.
+#' @param seek Character vector, indicating where `pattern` is sought. Use one
+#' of more of the following options:
+#'
+#' - `"name"`: searches in column names.
+#' - `"labels"`: searches in variable labels. Only applies when a `label` attribute
+#'   is set for a variable.
+#' - `"values"`: searches in value labels or factor levels. Only applies when a
+#'   `labels` attribute is set for a variable, or if a variable is a factor.
+#' - `"all"`: searches in all of the above.
+#' @param fuzzy Logical, if `TRUE`, "fuzzy matching" (partial and close distance
+#' matching) will be used to find `pattern`.
+#'
+#' @return A data frame with three columns: the column index, the column name
+#' and - if available - the variable label of all matched variables in `data`.
+#'
+#' @examples
+#' # seek variables with "Length" in variable name or labels
+#' seek_variables(iris, "Length")
+#'
+#' # seek variables with "dependency" in names or labels
+#' # column "e42dep" has a label-attribute "elder's dependency"
+#' data(efc)
+#' seek_variables(efc, "dependency")
+#'
+#' # "female" only appears as value label attribute - default search is in
+#' # variable names and labels only, so no match
+#' seek_variables(efc, "female")
+#' # when we seek in all sources, we find the variable "e16sex"
+#' seek_variables(efc, "female", seek = "all")
+#'
+#' # typo, no match
+#' seek_variables(iris, "Lenght")
+#' # typo, fuzzy match
+#' seek_variables(iris, "Lenght", fuzzy = TRUE)
+#' @export
+seek_variables <- function(data, pattern, seek = c("names", "labels"), fuzzy = FALSE) {
+  # check valid args
+  if (!is.data.frame(data)) {
+    insight::format_error("`data` must be a data frame.")
+  }
+
+  # check valid args
+  seek <- intersect(seek, c("names", "labels", "values", "levels", "column_names", "columns", "all"))
```

Comment:
"levels", "column_names", and "columns" are not listed in the possible choices in the docs

Reply:
Yes, that's intentional. I always forget the options, so I added some undocumented aliases for me :-D

Reply:
alright, might be worth adding a comment about this